sympy/solvers/simplex.py

```diff
@@ -0,0 +1,966 @@
+
+    Example
+    =======
+
+    Suppose you want to maximize
+        x
+    subject to
+        x >= 10
+        x <= 9
+
+    No x can satisfy those constraints.
+    """
+    pass
+
+
+def _pivot(M, i, j):
+    """
+    The pivot element `M[i, j]` is inverted and the rest of the matrix
+    modified and returned as a new matrix; original is left unmodified.
+
+    Example
+    =======
+
+    >>> from sympy.matrices.dense import Matrix
+    >>> from sympy.solvers.simplex import _pivot
+    >>> from sympy import var
+    >>> Matrix(3, 3, var('a:i'))
+    Matrix([
+    [a, b, c],
+    [d, e, f],
+    [g, h, i]])
+    >>> _pivot(_, 1, 0)
+    Matrix([
+    [-a/d, -a*e/d + b, -a*f/d + c],
+    [ 1/d,        e/d,        f/d],
+    [-g/d,  h - e*g/d,  i - f*g/d]])
+    """
+    Mi, Mj, Mij = M[i,:], M[:,j], M[i,j]
+    if Mij == 0:
+        raise ZeroDivisionError(
+            "Tried to pivot about zero-valued entry.")
+    A = M - Mj * (Mi / Mij)
+    A[i, :] = Mi / Mij
+    A[:, j] = -Mj / Mij
+    A[i, j] = 1 / Mij
+    return A
+
+
+def _choose_pivot_row(A, B, candidate_rows, pivot_col, Y):
+    # Choose row with smallest ratio
+    first_row = candidate_rows[0]
+    min_ratio = B[first_row] / A[first_row, pivot_col]
+    min_rows = [first_row]
+    for i in candidate_rows[1:]:
+        ratio = B[i] / A[i, pivot_col]
+        if ratio < min_ratio:
+            min_ratio = ratio
+            min_rows = [i]
+        elif ratio == min_ratio:
+            min_rows.append(i)
+
+    # If there are ties, pick using Bland's rule
+    row = sorted(min_rows, key= lambda r: Y[r])[0]
+    return row
+
+
+def _simplex(A, B, C, D=None, dual=False):
+    """
+    Return ``(o, x, y)`` obtained from the two-phase simplex method
+    using Bland's rule where ``o`` is the minimum value of primal,
+    ``Cx - D``, under constraints ``Ax <= B`` (with ``x >= 0``) and
+    the maximum of the dual, ``y^{T}B - D``, under constraints
+    ``A^{T}*y >= C^{T}`` (with ``y >= 0``). To compute the dual of
+    the system, pass `dual=True` and ``(o, y, x)`` will be returned.
+
+    Note: the nonnegative constraints for ``x`` and ``y`` supercede
+    any values of ``A`` and ``B`` that are inconsistent with that
+    assumption, so if a constraint of ``x >= -1`` is represented
+    in ``A`` and ``B``, no value will be obtained that is negative; if
+    a constraint of ``x <= -1`` is represented, an error will be
+    raised since no solution is possible.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.simplex import _simplex
+    >>> from sympy import Matrix
+
+    Consider the simple minimization of ``f = x + y + 1`` under the
+    constraint that ``y + 2*x >= 4``. In the nonnegative quadrant,
+    this inequality describes a area above a triangle with vertices at
+    (0, 4), (0, 0) and (2, 0). The minimum of ``f`` occurs at (2, 0).
+    Define A, B, C, D for the standard minimization:
+
+    >>> A, B, C, D = [Matrix(i) for i in [[[2, 1]], [4], [[1, 1]], [-1]]]
```

Comment:
I think it would be clearer to state these explicitly like `A = Matrix([[2, 1]])`, `B = Matrix(...)` rather than using a list comprehension.

Reply:
Maybe this could even show the inequalities to clarify the relationship between the matrices and the problem:
```python
In [5]: M = Matrix([[1, 2], [3, 4]])

In [6]: x = Matrix(symbols('x:2'))

In [7]: b = symbols('y:2')

In [8]: [a >= b for a, b in zip(M*x, b)] # No way to avoid a list comprehension here...
Out[8]: [x₀ + 2⋅x₁ ≥ y₀, 3⋅x₀ + 4⋅x₁ ≥ y₁]
```

Reply:
done